reporting/src/main/java/org/wso2/testgrid/reporting/PerformanceResultProcessor.java

```diff
@@ -0,0 +1,240 @@
+/*
+* Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.testgrid.reporting;
+
+import org.wso2.testgrid.common.Column;
+import org.wso2.testgrid.common.ResultFormatter;
+import org.wso2.testgrid.common.TestScenario;
+import org.wso2.testgrid.reporting.model.performance.ColumnHeader;
+import org.wso2.testgrid.reporting.model.performance.DataSection;
+import org.wso2.testgrid.reporting.model.performance.DividerSection;
+import org.wso2.testgrid.reporting.model.performance.PerformanceTable;
+import org.wso2.testgrid.reporting.model.performance.ScenarioSection;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Stack;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains the functionalities required to process the CSV result file and genreate the
+ * Report data model.
+ *
+ * @since 1.0.0
+ */
+public class PerformanceResultProcessor {
+
+    private static final String RESULT_FILE = "summary.csv";
+    private static final String GRAPH_EXTENSION = ".png";
+    private static final String IMAGES_FOLDER = "img";
+
+    /**
+     * This method returns the {@link ScenarioSection} that is processed from the TestScenario ob object
+     *
+     * @param testScenario TestScenario being processed
+     * @param resultFormatter ResultFormatter containing the format data
+     * @return ScenarioSection object that is included in the report
+     */
+    public ScenarioSection processScenario(TestScenario testScenario, ResultFormatter resultFormatter) {
+        List<List<String>> data = testScenario.getPerformanceTestResults();
+        ScenarioSection scenarioSection = new ScenarioSection();
+        scenarioSection.setScenarioName(testScenario.getName());
+        scenarioSection.setDescription(testScenario.getDescription());
+        List<String> topics = data.remove(0);
```

Comment:
It's not good to modify the testScenario's performanceTestResults object. So, I think testScenario.getPerformanceTestResults() should not return the actual internal List instance. It should rather create a copy and return that. 

Ref: https://stackoverflow.com/a/15778698/388714